common/reactive-grpc-common/src/main/java/com/salesforce/reactivegrpc/common/AbstractStreamObserverAndPublisher.java

```diff
@@ -249,20 +234,17 @@ void drainFused(final Subscriber<? super T> subscriber) {
             }
 
             missed = WIP.addAndGet(this, -missed);
-            if (missed == 0) {
-                break;
-            }
-        }
+        } while (missed != 0);
     }
 
-    void drain() {
+    private void drain() {
         if (WIP.getAndIncrement(this) != 0) {
             return;
         }
 
         int missed = 1;
 
-        for (;;) {
+        do {
```

Comment:
Let's keep it as in the original code. From my point of view it preserves readability

Reply:
Fixed